### DISCUSSION TITLE
Migration guidance

### DISCUSSION THREAD

**Discussion by SzymonPobiega** [rabbitmq/rabbitmq-website#1675] [General]
### Is your feature request related to a problem? Please describe.

There could be a document in the *guidance* section describing how to migrate your distributed system from a legacy queuing technology e.g. MSMQ to RabbitMQ

### Describe the solution you'd like

A document with code snippets based on the [messaging bridge](https://www.enterpriseintegrationpatterns.com/patterns/messaging/MessagingBridge.html) pattern describing how a multi-endpoint system can be migrated one-endpoint-at-a-time to RabbitMQ with zero downtime

### Additional context

I can try to contribute such a document. I am the author of an open source implementation of the messaging bridge pattern https://github.com/SzymonPobiega/NServiceBus.Router.

Disclaimer: I also happen to work for Particular Software

**Comment by michaelklishin** [comment]
Our team does not have the resources to produce a migration guide for every, or even half a dozen most popular, messaging systems that exist today. In particular those that require a commercial license to run.

**Comment by michaelklishin** [comment]
To be honest, I think it makes more sense for NServiceBus to document what this migration tool can do, or publish a series of blog posts.

RabbitMQ does not have much "vendor-specific" documentation on the site. I don't want to speculate if it were the case if our team was larger and had enough folks to dedicate to a significantly larger documentation effort.

This project seems interesting to me but I'd say, since you know how to use it best, and it is a vendor-specific solution, it belongs to the blog of NService Bus, Particular Software, or your personal blog.

Our team won't be able to produce a series of posts as good as you can.

**Comment by SzymonPobiega** [comment]
Thanks for the speedy response @michaelklishin . I didn't mean to make it vendor-specific content. I already contributed something related to [MS docs](https://github.com/MicrosoftDocs/architecture-center/pull/4068#issuecomment-1578208216).

My idea was to have snippets to receive/send messages in both MSMQ and RabbitMQ SKDs to show how messaging systems could be migrated in a peace meal fashion. 

But I do understand you point with regards to the number of messaging technologies that could be covered in such an article. It was my assumption that the MSMQ is one of the most technologies people would want to migrate from to RabbitMQ and that assumption is based on the conversation with developers at various conferences.
